ytfzf

```diff
@@ -2271,6 +2274,27 @@ do_an_event_function "on_post_set_vars"
 : "${ytdl_pref:=$video_pref+$audio_pref/best/$video_pref/$audio_pref}"
 : "${shortcut_binds="Enter,double-click,${next_page_shortcut},${download_shortcut},${video_shortcut},${audio_shortcut},${detach_shortcut},${print_link_shortcut},${show_formats_shortcut},${info_shortcut},${search_again_shortcut},${custom_shortcut_binds},${custom_shortcut_binds}"}"
 
+# URL for instance list sorted by type, health and api (ones without API do not work!)
+instances_url="https://api.invidious.io/instances.json?sort_by=type,health,api"
+
+# Where do we put the list of healthy instances?
+: "${instances_file:="$cache_dir/instances.json"}"
+
+# If file doesn't already exist (or if force-refresh is requested), cURL and cache it.
+# CHECK: implement check for force-request
+# CHECK: added --refresh-inv-instances to optargs
+# TODO: update docs
+[ ! -f "$instances_file" ] ||
+    [ "$refresh_inv_instances" -eq 1 ] &&
```

Comment:
`refresh_inv_instances` should be set to 0 in the `set_vars` function, otherwise there's an error when `--rii` ins't givent.

Reply:
Alright, added that in too.
I've added some options to the `curl` command, to avoid overwriting cache with an empty file, in case of internet issues (which is what inspired that fix too).

Reply:
This seems to work well, nice addition